app/src/main/java/com/github/featuredetectandroid/ui/AppLayout.kt

```diff
@@ -0,0 +1,81 @@
+package com.github.featuredetectandroid.ui
+
+import android.content.Context
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.PointMode
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.LocalDensity
+import com.github.featuredetectandroid.utils.KeypointDetectionAlgorithm
+import com.github.featuredetectandroid.utils.PreferencesManager
+import com.github.featuredetectandroid.utils.selectFeatureDetector
+
+@Composable
+fun AppLayout(
+    context: Context,
+    outputViewModel: OutputViewModel,
+    preferencesManager: PreferencesManager,
+    selectedAlgorithm: String,
+    onAlgorithmSelected: (String) -> Unit
+) = Scaffold(
+    drawerContent = {
+        Menu(
+            header = "Keypoint detection algorithm:",
+            options = KeypointDetectionAlgorithm.names,
+            selectedOption = selectedAlgorithm,
+            onSelected = { algorithmName ->
+                preferencesManager.putSelectedAlgorithm(algorithmName)
+                onAlgorithmSelected(algorithmName)
+                outputViewModel.keypointOffsets = emptyList()
+                outputViewModel.featureDetector = selectFeatureDetector(
```

Comment:
why does viewmodel require context?

Reply:
This is needed for the `SuperPoint` constructor, now I also handle this in `onDestroy` and `onResume`.

Reply:
You don't need to do anything like this in `onDestroy` and `onResume` as `SuperPoint` doesn't hold any links to `Context` inside it. You can leave a comment where you pass the `Context` so that it won't be forgotten. I'll also add it to `SuperPoint`'s docs.

Reply:
fixed